homeassistant/components/conversation/default_agent.py

```diff
@@ -121,6 +121,30 @@ async def async_process(
             response=intent_response, conversation_id=conversation_id
         )
 
+    async def async_reload(self, language: str | None = None):
+        """Clear cached intents for a language."""
+        if language is None:
+            language = self.hass.config.language
+
+        self._lang_intents.pop(language, None)
+        _LOGGER.debug("Cleared intents for language: %s", language)
+
+    async def async_prepare(self, language: str | None = None):
+        """Load intents for a language."""
+        if language is None:
+            language = self.hass.config.language
+
+        lang_intents = await self.hass.async_add_executor_job(
```

Comment:
we should introduce a per-language lock.

It can now happen that our prepare call is still computing while the actual sentence comes in, and it's loading the same intents twice.